inference-engine/src/legacy_api/include/legacy/details/ie_cnn_network_iterator.hpp

```diff
@@ -98,16 +98,31 @@ CNNNetworkIterator {
             }
         };
 
+        // Find all outputLayers
+        std::vector<CNNLayerPtr> outputLayers;
+        const auto* networkImpl = dynamic_cast<const CNNNetworkImpl*>(network);
+        if (networkImpl) {
+            for (const auto & node : networkImpl->allLayers()) {
```

Comment:
Could you explain why do we need to extend output layers with all other nodes? Looks very strange.

Reply:
I  use this cycle in order to find all output and memory layers